tensorflow/lite/micro/micro_allocation_info.cc

```diff
@@ -240,6 +240,7 @@ TfLiteStatus AllocationInfoBuilder::MarkAllocationLifetimes(
     const int tensor_index = subgraph->inputs()->Get(i);
     AllocationInfo* current = &subgraph_allocation_info[tensor_index];
     UpdateFirstCreated(current, allocation_scope_count_);
+    UpdateLastUsed(current, allocation_scope_count_);
```

Comment:
Thanks for the PR and the detailed description, Cedric! 
Looking at the example in the description, that "unused" tensor for the WHILE OP would be marked with last used after the two subgraphs have been traversed (Specifically at line 276).  The proposed change here does not hurt but should not help as well because line 276 should kick in and override the last_used. 

Note that there is a TestMultiSubgraphWithIf in micro_allocation_info_test.cc that test some similar scenarios that you describe. Maybe can craft a similar unit test case?

The memory planner code is tricky and I could be missing something here; so feel free to let us know what you think.

Reply:
Thank you for the review and apologies for the late reply. I've now slightly modified the existing test model (I thought copy-pasting 99% of the test would not be good practice) to no longer use 2 inputs for the `MUL` operator in the else-branch:
![image](https://user-images.githubusercontent.com/1162108/186682045-20302026-c20a-4c57-8d83-e2b923a739b1.png)


Then, I undid my change again and it resulted in this assert:
```
allocation_info[8].last_used == 4 failed at tensorflow/lite/micro/micro_allocation_info_test.cc:118 (-1 vs 4)
```
In other words, without my modification, this input to the then-graph is considered unused. However (at least in the `while.cc` implementation) it is used, even if none of the ops in the graph use it (like in the Python example above).

Note that the reason the values are now 4 instead of 5 is that they are no longer used in the `MUL` operation.

So I am now more confident that my fix is indeed a proper fix. Let me know what you think.